lib/serviceClient.ts

```diff
@@ -384,25 +384,23 @@ export function getPropertyParent(parent: PropertyParent, propertyPath: string[]
   return parent;
 }
 
-function getOperationArgumentValueFromParameter(operationArguments: OperationArguments, parameter: OperationParameter, serializer: Serializer): any {
-  return getOperationArgumentValueFromParameterPath(operationArguments, parameter.parameterPath, parameter.mapper, serializer);
+function getOperationArgumentValueFromParameter(serviceClient: ServiceClient, operationArguments: OperationArguments, parameter: OperationParameter, serializer: Serializer): any {
+  return getOperationArgumentValueFromParameterPath(serviceClient, operationArguments, parameter.parameterPath, parameter.mapper, serializer);
 }
 
-function getOperationArgumentValueFromParameterPath(operationArguments: OperationArguments, parameterPath: ParameterPath, parameterMapper: Mapper, serializer: Serializer): any {
+function getOperationArgumentValueFromParameterPath(serviceClient: ServiceClient, operationArguments: OperationArguments, parameterPath: ParameterPath, parameterMapper: Mapper, serializer: Serializer): any {
   let value: any;
   if (typeof parameterPath === "string") {
     parameterPath = [parameterPath];
   }
   if (operationArguments.arguments) {
     if (Array.isArray(parameterPath)) {
       if (parameterPath.length > 0) {
-        value = operationArguments.arguments;
-        for (const parameterPathPart of parameterPath) {
-          value = value[parameterPathPart];
-          if (value == undefined) {
-            break;
-          }
+        let propertySearchResult: PropertySearchResult = getPropertyFromParameterPath(operationArguments.arguments, parameterPath);
```

Comment:
This feels like a nice place for `const { propertyFound, propertyValue } = ...`

Reply:
Can't do that here, because I may reassign propertySeachResult in two lines.